banyand/metadata/embeddedserver/server.go

```diff
@@ -103,3 +139,38 @@ func NewService(_ context.Context) (metadata.Service, error) {
 	}
 	return s, nil
 }
+
+func (s *server) RegisterDefrag(ctx context.Context) {
+	var (
+		err        error
+		etcdLogger = logger.GetLogger().Named("etcd-server")
+		defrag     = func(_ time.Time, _ *logger.Logger) bool {
+			for _, endpoint := range s.listenClientURL {
+				_, err = s.ecli.Defragment(ctx, endpoint)
+				if err != nil {
+					etcdLogger.Error().Err(err).Msg("failed to execute defragmentation")
+					return false
+				}
+			}
+			return true
+		}
+	)
+
+	if s.ecli == nil {
```

Comment:
```suggestion
```

If `s.ecli` is nil, it indicates that there is a bug present. You should notify it immediately.